src/components/Header/Header.tsx

```diff
@@ -19,57 +28,130 @@ const useStyles = makeStyles<Theme>(theme => ({
+  },
+  roundedShape: {
+    borderRadius: '0 0 100% 100%',
+  },
+  roundedAppBarBackground: {
+    width: '110%',
+    padding: '0 5%',
+    marginLeft: '-5%',
+  },
+  roundedLoader: {
+    height: '101%',
+  },
+  roundedAppBarContainer: {
+    overflow: 'hidden',
+    width: '100%',
+    height: '100%',
+    position: 'sticky',
+    top: 0,
+    right: 0,
+    left: 'auto',
+    zIndex: theme.zIndex.appBar,
+    paddingBottom: '1%',
   },
 }));
 
 const Header: React.FC<HeaderProps> = ({
   type,
-  color = 'default',
+  color = MenuColors.DEFAULT,
+  shape = MenuShapes.DEFAULT,
   children,
+  loading,
   onButtonClick,
 }) => {
-  const classes = useStyles();
-  const scrollTrigger = useScrollTrigger({ threshold: 10 });
+  const {
+    menuButton,
+    appBar,
+    withoutShadow,
+    defaultAppBarBackground,
+    paperAppBarBackground,
+    loaderShadow,
+    backgroundLayer,
+    roundedShape,
+    roundedAppBarBackground,
+    roundedLoader,
+    roundedAppBarContainer,
+  } = useStyles();
+
+  const scrollTrigger = useScrollTrigger({ disableHysteresis: true, threshold: 10 });
+  const isRoundedShape = shape === MenuShapes.ROUNDED;
 
   const button: NavBarButton = {
-    menu: () => <MenuIcon />,
-    back: () => <ArrowBackIcon />,
+    [MenuTypes.MENU]: () => <MenuIcon />,
+    [MenuTypes.BACK]: () => <ArrowBackIcon />,
   };
 
   const colorClassNames = {
-    default: classes.defaultAppBarBackground,
-    paper: classes.neutralAppBarBackground,
+    [MenuColors.DEFAULT]: defaultAppBarBackground,
+    [MenuColors.PAPER]: paperAppBarBackground,
+  };
+
+  const shapeClassNames = {
+    [MenuShapes.DEFAULT]: '',
+    [MenuShapes.ROUNDED]: [roundedAppBarBackground, roundedShape],
   };
 
-  return (
+  const HeaderBar = () => (
     <AppBar
       component={'header'}
-      className={clsx(
-        classes.appBar,
-        {
-          [classes.withoutShadow]: !scrollTrigger,
-        },
-        colorClassNames[color]
-      )}
+      className={clsx(appBar, colorClassNames[color], shapeClassNames[shape], {
+        [withoutShadow]: loading || !scrollTrigger,
+      })}
       position={'sticky'}
       color={'inherit'}>
       <Toolbar>
         <IconButton
           edge='start'
-          className={classes.menuButton}
+          className={menuButton}
           color='inherit'
           onClick={onButtonClick}>
           {button[type]()}
         </IconButton>
       </Toolbar>
       {children}
+      {loading && (
+        <LinearProgress
+          variant={'indeterminate'}
+          className={clsx(
+            loaderShadow,
+            isRoundedShape && [roundedShape, roundedLoader, backgroundLayer]
+          )}
+        />
+      )}
+      <Box
+        className={clsx(
+          paperAppBarBackground,
+          isRoundedShape && [roundedShape, backgroundLayer]
+        )}
+      />
     </AppBar>
   );
+
+  return isRoundedShape ? (
+    <Box className={roundedAppBarContainer}>
```

Comment:
I found out that header is occupying more space than it needs. Take a look

<img width="300" alt="Screenshot 2019-12-16 at 09 08 18" src="https://user-images.githubusercontent.com/12997200/70893897-e51ad680-1fe3-11ea-9141-2de3c5c4ab11.png">